keps/sig-apps/20190318-PodDisruptionBudget-graduation-to-stable.md

```diff
@@ -0,0 +1,133 @@
+---
+title: graduate-pod-disruption-budget-to-stable
+authors:
+  - "@bsalamat"
+owning-sig: sig-apps
+participating-sigs:
+  - sig-scheduling
+reviewers:
+  - "@liggitt"
+  - "@kow3ns"
+approvers:
+  - "@kow3ns"
+  - "@liggitt"
+editor: TBD
+creation-date: 2019-03-18
+last-updated: 2019-03-18
+status: implementable
+see-also:
+  - 
+replaces:
+superseded-by:
+---
+
+# Graduate PodDisruptionBudget to stable
+
+## Table of Contents
+
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints-optional)
+  - [Risks and Mitigations](#risks-and-mitigations)
+  - [Test Plan](#test-plan)
+  - [Graduation Criteria](#graduation-criteria)
+- [Implementation History](#implementation-history)
+
+## Summary
+
+[Pod Disruption Budget (PDB)](https://kubernetes.io/docs/tasks/run-application/configure-pdb/)
+is a Kubernetes API that limits the number of pods of a collection that are down simultaneously from voluntary disruptions.
+[Kubernetes eviction API](https://kubernetes.io/docs/tasks/administer-cluster/safely-drain-node/#the-eviction-api)
+takes PDB into account when terminating pods. If PDB is
+violated, the eviction API returns failure and does not delete the requested pod.
+The feature was introduced in Kubernetes 1.4 and promoted to beta in 1.5.
+It has been in beta for a long time. This document lays out the plan to promote
+it to stable.
+
+## Motivation
+
+PDB API has been stable and is an important feature that allows users to improve
+reliability of their critical workloads. This feature has been in beta for a
+long time. We need to promote to stable version given that we plan to support it
+long term.
+
+### Goals
+
+* Plan to promote PDB API to stable version.
+* Propose changes to the deployment controller to take PDB into account.
```

Comment:
Where is "taking it into account" described?

Reply:
I removed that section as it turns out to create [complexities](https://github.com/kubernetes/enhancements/pull/904#discussion_r266709445) which are not easy to address. I'll remove this goal.